internal/request/graphql/parser.go

```diff
@@ -50,7 +50,7 @@ func NewParser() (*parser, error) {
 }
 
 func (p *parser) BuildRequestAST(ctx context.Context, request string) (*ast.Document, error) {
-	_, span := tracer.Start(ctx)
+	ctx, span := tracer.Start(ctx) //nolint:ineffassign,staticcheck
```

Comment:
todo: Why do you need to do this. 1) `ctx` is not used anywhere and 2) it is ineffectual assignment.

I like what you had before.

Same for all other instances

Reply:
He change it because Andy asked him to do so I think. I'm also not a fan of this change.

Reply:
Didn't seem like a strong preference by him, I commented on the original thread now.